src/hooks/types.rs

```diff
@@ -0,0 +1,15 @@
+use std::collections::HashMap;
+
+use crate::{evaluation::EvaluationContext, providers::types::ProviderMetadata, ClientMetaData};
```

Comment:
Casing of `Metadata` is inconsistent. I'd recommend it's a single word, but I think it should be consistent, at a minimum.

Reply:
Updated in next commit